### ISSUE TITLE
Download Subpages

### ISSUE THREAD

**Issue by javierarce** [WordPress/wporg-main-2022#129]
Here's the proposed design for the download subpages: Releases, Beta/Nightly, Source Code, Counter that @beafialho, @critterverse, and I have worked on [this file](https://www.figma.com/file/WHhRwVc4swYnPKdyVWj3du/Home-%26-Download-Pages?node-id=2093%3A22538).

I'll share the proposals for the desktop and mobile pages and mention some aspects of the design that could be categorized as 'nice to have'. At the end of the issue I'll add some notes about the navigation of these internal pages:

### Releases 

![image](https://user-images.githubusercontent.com/4933/192808301-1d296cb7-38e8-413b-965c-1542ca298279.png)

![image](https://user-images.githubusercontent.com/4933/192808545-3e6b00d0-7e51-4aea-a27c-cc307a0016f7.png)

![image](https://user-images.githubusercontent.com/4933/192808644-d2ab81bb-9e94-48df-b768-ab5f7ae187ce.png)

### Nice to have

- Instead of listing every release on the same page, older releases could be accessed through a list at the bottom of the page. We could either load each table under the Past releases section or have a separate page per version.
- Each major release is identified with both the version number and its code name.

---

### Beta/Nightly

![image](https://user-images.githubusercontent.com/4933/192808808-aee85f3c-7a93-4a37-af55-bbf0dd858e2d.png)

---

### Counter

![image](https://user-images.githubusercontent.com/4933/192815889-4267f152-b754-404c-a954-8f637bfd43e1.png)

Something we could explore on this page is a smoother transition between numbers. We could also review the font we chose and use a monospaced one so the numbers stay in the same position as the numbers increase.

---

### Source code

![image](https://user-images.githubusercontent.com/4933/192808959-82b380e7-a544-46ba-9833-dc90859ae027.png)

---

### Navigation

![image](https://user-images.githubusercontent.com/4933/192815803-1f4c2ea5-be5a-4c02-9577-dfe6b51aa798.png)

The other challenging aspect of these designs are the subnavigations: on smaller viewports our proposal is for the right-side lists to collapse into dropdowns.

---

[Figma file](https://www.figma.com/file/WHhRwVc4swYnPKdyVWj3du/Home-%26-Download-Pages?node-id=2093%3A22538) with the designs.

**Comment by ryelle** [comment]
@javierarce Just confirming next steps, since you've labelled this "needs design feedback" and "needs copy feedback" — none of this is ready to be built yet, correct?

**Comment by javierarce** [comment]
Not yet, but I think it should be pretty close: I'd love to have more eyes on the designs in case we have missed anything important or in case we could refine the designs/copy a little bit. 

I also like to know if those "nice to have" things I mentioned can be implemented or need to be simplified.

**Comment by ryelle** [comment]
This looks good! I've responded to the nice-to-have comments and left some more technical details below. The questions I've added are for other devs with more historical background :)

**Releases**

> Instead of listing every release on the same page, older releases could be accessed through a list at the bottom of the page. We could either load each table under the Past releases section or have a separate page per version.

Looking at how this is currently rendered, I think it will make the most sense to just port the existing code into one single dynamic block, to render all the headings & tables. I’m sure we can make something work with toggling CSS classes or something for showing single tables at a time.

> Each major release is identified with both the version number and its code name.

We don’t have the code names associated with the releases in the current tables. Can we get that from somewhere else in the code?

**Beta/Nightly**

No concerns, actually put this together on WordPress.org in about 5 min 🙂

**Counter**

The counter should be a custom block, with support for UI controls so we can try out monospace fonts etc.

The counter page also provides `https://wordpress.org/download/counter/?ajaxupdate=1`, used for loading in the new number, and `https://wordpress.org/download/counter/?json=1` - where is this used - is it a “public” API? Do we need to stick to this functionality?

**Source code**

Also seems like a straightforward page.

**Navigation**

We should look at how the nav bar was built in News for this, we might be able to reuse some of that code. Can probably create a block style on the group block to add the brush stroke (IIRC this is a mask, so the background of the group would still be customizable).

I think functionally we can use the core navigation’s mobile toggle for the small-screen behavior, but it’ll need custom styling to avoid overlaying the whole page.

How I see this working:

```
- Group block with brush style, full width
  - Row block, wide width, space between alignment
    - Paragraph, "Download"
    - Navigation block with page list
```

And I think we would also create a pattern for designers to use on the other pages (I imagine a similar setup will be used for About subpages, for example).

```
- Group block with brush style, full width
  - Row block, wide width, space between alignment
    - Paragraph, "Section title"
    - Empty Navigation block
```

----

Once this is ready for implementation, we can break this down into individual issues for the counter block, releases block, group brush stroke style, and section navigation.

**Comment by dd32** [comment]
> **Releases**

I think some feedback might be deserved here on whether we need to link to the `IIS` builds here, AFAIK they are no longer actively used, as they were designed for a time where we submitted WordPress ZIPs to Microsoft for use in their App Installer, if I'm recalling it correctly. 

> > Each major release is identified with both the version number and its code name.
> 
> We don’t have the code names associated with the releases in the current tables. Can we get that from somewhere else in the code?

We don't store that detail in code anywhere, the only data-store used for this page is the files on disk itself, which has lead to some bugs such as https://meta.trac.wordpress.org/ticket/4298

I think we can probably resolve this by storing extra details on disk with the release archives, specifying the codename, date, and possibly the announcement post? (I personally would like to see announcement posts linked from that page)

> **Counter**
>
> The counter page also provides [...]  `https://wordpress.org/download/counter/?json=1` - where is this used - is it a “public” API? Do we need to stick to this functionality?

I've never noticed that API endpoint, and when initially asked assumed it was the ajax update endpoint. Looking through SVN history, it appears that it was introduced at the same time as the `?ajaxupdate` endpoint - potentially as an alternative implementation.
I personally would suggest we treat both this API and the `?ajaxupdate` API as internal, and don't concern ourselves with back-compat for it. If it turns out that someone is actually using it, and they can't migrate to a new one, then we can reconsider this.

**Comment by jasmussen** [comment]
Visually this looks great to me. Tables are glancable, main text column with code font and spacing look coherent, subnavigation looks clean and consistent. 

I love the full-bleed download page. We'll probably want to do something similar on this page as was done to the /mobile page, where a container is something a la 80vh tall and vertically centers the content inside, with some padding.

There's something about the vertical lines here, where "Download" could left align with "News":

<img width="537" alt="Screenshot 2022-09-29 at 09 41 07" src="https://user-images.githubusercontent.com/1204802/192970785-ac910bb0-f67d-4a29-ad8d-de9ef4638119.png">

In this particular full-bleed design, it also becomes clearer that the "Download" subheading is a larger font than the menu items. But only a little larger, so it's lacking some size contrast. Are "Releases" etc, on the right the same font size as "Download"? Can we try them the same size as the top menu items? If need be, "Download" can be bold.

**Comment by javierarce** [comment]
> There's something about the vertical lines here, where "Download" could left align with "News"

We could try to do that, but we'll probably need to update some subnavigations in other places. In News, for example, the left and right padding are dynamic and they are also used in some layout elements. 

I'd suggest addressing this after we have implemented these pages in a separate issue so we can keep a consistent behaviour. 

https://user-images.githubusercontent.com/4933/192997612-a58a070c-57fe-4264-bedc-b20237acf548.mov

> Are "Releases" etc, on the right the same font size as "Download"? Can we try them the same size as the top menu items?

Yes, we are using `16px` for all the items of that subnavigation. This also comes from the News nav style, so if we make this change we should bring it to those pages too.

> If need be, "Download" can be bold.

I think this is a better solution than setting the same font size for the navigation and subnavigation elements, as I imagine the original intention with making the subnav items bigger was to give it more presence.

Also, we use bold on internal pages like [this one](https://wordpress.org/news/category/awards)

> I think some feedback might be deserved here on whether we need to link to the IIS builds here, AFAIK they are no longer actively used, as they were designed for a time where we submitted WordPress ZIPs to Microsoft for use in their App Installer, if I'm recalling it correctly.

@dd32 this would give us more space to present the other two download links, which would be very nice… how should we proceed here to remove those links? Should we ask the community first?

> I think we can probably resolve this by storing extra details on disk with the release archives, specifying the codename, date, and possibly the announcement post? (I personally would like to see announcement posts linked from that page)

Linking the announcement posts would be fantastic. We have all that information [here](https://wordpress.org/about/history). Maybe if we add this information to the download page we won't need the history one.

@ryelle thanks a lot for your analysis of the technical aspects of the implementation!

**Comment by beafialho** [comment]
> Yes, we are using 16px for all the items of that subnavigation. This also comes from the News nav style, so if we make this change we should bring it to those pages too.

> I imagine the original intention with making the subnav items bigger was to give it more presence.

I understand the intent in aligning the font sizes in both navigations and completely support that if you think it works better throughout the whole site. My suggestion would be to make "Download" bold and see if that looks less awkward.

> There's something about the vertical lines here, where "Download" could left align with "News":

This is happening because the previous version of the header had the "WordPress.org" word mark next to the logo, which made it look like it had independent alignments. This one actually looks like a misalignment.

<img width="773" alt="Captura de ecrã 2022-09-29, às 12 59 36" src="https://user-images.githubusercontent.com/65220155/193026142-f6765fde-74d5-48ea-9f91-9fe85708fe4c.png">

**Comment by javierarce** [comment]
> My suggestion would be to make "Download" bold and see if that looks less awkward.

Perfect, I'll explore this.

> This is happening because the previous version of the header had the "WordPress.org" word mark next to the logo, which made it look like it had independent alignments. This one actually looks like a misalignment.

Ok, I'll open an issue so we can address this globally.

**Comment by javierarce** [comment]
I've been digging a bit more into the alignment issue and this is what I found:

I think adjusting that alignment (represented by the solid lines in the image below) would also require modifyng the left and right padding of the content and the footer.

![image](https://user-images.githubusercontent.com/4933/193057292-55164d4e-47d1-4e55-b448-f564807588e1.png)

I did a quick test setting `--wp--custom--alignment--edge-spacing` to `clamp( 24px, calc( 100vw ), 85px );` and this is what an aligned version could look like (we would lose some spacing around the left and right sides).

![wordpress org_news_2022_09_wordpress-6-1-beta-2-now-available_ (1)](https://user-images.githubusercontent.com/4933/193058902-bb438380-ddb7-4809-8c4d-44e1fa7889a2.png)

**Comment by ryelle** [comment]
The way the alignment works on News is different from how it works on the new WP.org theme, since they're not actually using the same theme (or `theme.json`). I don't think we have any plans to update News (except for fixing bugs).

This is how I had interpreted your mockup working at larger sizes, since this is how the other WordPress.org pages work:

![Screen Shot 2022-09-29 at 10 46 52](https://user-images.githubusercontent.com/541093/193064441-f5da9375-6fa6-492e-a6f4-58684efea2b4.png)

We could full-width that top bar and use padding, but we're limited to what we can set in Gutenberg ([the values here](https://github.com/WordPress/wporg-parent-2021/#spacing), or a fixed size).

![Screen Shot 2022-09-29 at 11 10 56](https://user-images.githubusercontent.com/541093/193070054-8bf58b61-57ec-4e14-baef-9367187e1699.png)

Screenshots are from the Beta/Nightly page on wporg if you want to jump in and try different alignments. It won't affect anything to update the content there, since we haven't set that page up for syncing yet.

**Comment by javierarce** [comment]
Thanks, @ryelle! If News is not going to change in the foreseeable future, the best solution for this sub-navigation would be to make the alignment of these new navigations as close as the ones in News as possible.

In any case, I'm going on holiday for a week and a half, if no one does it before I'm back I'll give it a spin then.

**Comment by thetinyl** [comment]
A few notes/thoughts around copy for consideration:

For consistency, I'd recommend sticking to sentence case for headlines/subheadlines if possible to match what's been done on the Homepage/Download page (exception for /News article titles).

For the Releases page, I wonder if there's an opportunity to tighten the language/take the "we" out (since it's more product, less community focused). Suggested edits below in bold:

> This is an archive of every release **that’s been recorded.** **Only the latest in the 6.0 series is actively maintained and safe to use.** 
>
> Curious about which jazzers **were** highlighted for each release? **Check out the full list.**

**Comment by javierarce** [comment]
> For the Releases page, I wonder if there's an opportunity to tighten the language/take the "we" out (since it's more product, less community focused). Suggested edits below in bold:

Thanks for the suggestions, @thetinyl! I've updated the copy on the [Figma file](https://www.figma.com/file/WHhRwVc4swYnPKdyVWj3du/Home-%26-Download-Pages?node-id=2093%3A22538). 

If we end up adding the codenames I think we won't need the last sentence ("Curious about which jazzers…") so I removed it.

**Comment by ryelle** [comment]
What still needs design decisions here? Could Neso at least start building the Counter & Release table blocks? We don't need finalized copy for dev - we can move the content into the respective pages, and let writers edit the content directly.

I _think_ the only outstanding issue is the Navigation header alignment, which would be best solved in the editor.

**Comment by jasmussen** [comment]
> I think the only outstanding issue is the Navigation header alignment, which would be best solved in the editor.

Yep, sounds good!

If there are any details, we can always come back in separate tickets. Thank you!
